docs/content/FormControl.mdx

```diff
@@ -0,0 +1,414 @@
+>
+  <FormControl disabled>
+    <FormControl.Label>Name</FormControl.Label>
+    <TextInput />
+  </FormControl>
+
+  <FormControl variant="choice" disabled>
+    <FormControl.Label>Checkbox option</FormControl.Label>
+    <Checkbox />
+  </FormControl>
+</Box>
+```
+
+### With a visually hidden label
+
+<Note>
+
+Every input must have a corresponding label to be accessible to assistive technology. That's why you'd use `FormControl` instead of using [`TextInput`](/TextInput) directly.
+
+`FormControl` also provides an interface for showing a hint text caption and a validation message, and associating those with the input for assistive technology.
+
+</Note>
+
+```jsx live
+<Box
+  display="flex"
+  flexDirection="column"
+  as="form"
+  sx={{
+    '> * + *': {
+      marginTop: 4
+    }
+  }}
+>
+  <FormControl>
+    <FormControl.Label visuallyHidden>Name</FormControl.Label>
+    <TextInput />
+  </FormControl>
+  <FormControl variant="choice">
+    <FormControl.Label visuallyHidden>Checkbox option</FormControl.Label>
+    <Checkbox />
+  </FormControl>
+</Box>
+```
+
+### With a caption
+
+```jsx live
+<Box
+  display="flex"
+  flexDirection="column"
+  as="form"
+  sx={{
+    '> * + *': {
+      marginTop: 4
+    }
+  }}
+>
+  <FormControl>
+    <FormControl.Label>Name</FormControl.Label>
+    <TextInput />
+    <FormControl.Caption>Hint: your first name</FormControl.Caption>
+  </FormControl>
+  <FormControl variant="choice">
+    <FormControl.Label>Option one</FormControl.Label>
+    <Checkbox />
+    <FormControl.Caption>Hint: the first and only option</FormControl.Caption>
+  </FormControl>
+</Box>
+```
+
+### With validation
+
+<Note>Validation messages are not used for an individual checkbox or radio form control.</Note>
+
+```javascript live noinline
+const ValidationExample = () => {
+  const [value, setValue] = React.useState('mona lisa')
+  const [validationResult, setValidationResult] = React.useState()
+  const doesValueContainSpaces = inputValue => /\s/g.test(inputValue)
+  const handleInputChange = e => {
+    setValue(e.currentTarget.value)
+  }
+
+  React.useEffect(() => {
+    if (doesValueContainSpaces(value)) {
+      setValidationResult('noSpaces')
+    } else if (value) {
+      setValidationResult('validName')
+    }
+  }, [value])
+
+  return (
+    <FormControl>
+      <FormControl.Label>GitHub handle</FormControl.Label>
+      <TextInput block value={value} onChange={handleInputChange} />
+      {validationResult === 'noSpaces' && (
+        <FormControl.Validation>GitHub handles cannot contain spaces</FormControl.Validation>
+      )}
+      {validationResult === 'validName' && <FormControl.Validation>Valid name</FormControl.Validation>}
+      <FormControl.Caption>With or without "@". For example "monalisa" or "@monalisa"</FormControl.Caption>
+    </FormControl>
+  )
+}
+
+render(ValidationExample)
+```
+
+### With a leading visual
+
+<Note>Only a checkbox or radio form control may have a leading visual.</Note>
```

Comment:
Actually, I think most of these notes could go after the code example and should maybe be marked as warnings?

Reply:
I had them marked as warnings at first, but it felt aggressive